SerialPrograms/Source/PokemonSV/Inference/PokemonSV_ESPEmotionDetector.cpp

```diff
@@ -0,0 +1,139 @@
+/*  ESP Emotion Detector
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/ImageTools/SolidColorTest.h"
+#include "CommonFramework/ImageTools/ColorClustering.h"
+#include "CommonFramework/VideoPipeline/VideoFeed.h"
+#include "CommonFramework/VideoPipeline/VideoOverlayScopes.h"
+#include "CommonFramework/Inference/InferenceThrottler.h"
+#include "PokemonSV_ESPEmotionDetector.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+ESPEmotionDetector::ESPEmotionDetector(Logger& logger, VideoOverlay& overlay)
+    : m_logger(logger)
+    , m_symbol_box(overlay, { 0.297, 0.137, 0.010, 0.016 }) //emotion
+{}
+
+ESPEmotionDetector::Detection ESPEmotionDetector::detect(const ImageViewRGB32& screen){
+    ImageViewRGB32 symbol = extract_box_reference(screen, m_symbol_box);
+
+    //Color ratio: R/(R+G+B), G/(R+G+B), B/(R+G+B)
+    if (is_solid(symbol, { 0.567, 0.2, 0.232 }, 0.1, 10)) {
+        return Detection::RED;
+    }
+    if (is_solid(symbol, { 0.529, 0.447, 0.0258}, 0.1, 10)) {
+        return Detection::YELLOW;
+    }
+    if (is_solid(symbol, { 0.132, 0.332, 0.536 }, 0.1, 10)) {
+        return Detection::BLUE; //Picks up the grey as well but that works
+    }
+    if (is_solid(symbol, { 0.323, 0.491, 0.184 }, 0.1, 10)) {
+        return Detection::GREEN;
+    }
+    if (is_solid(symbol, { 0.219, 0.355, 0.426 }, 0.1, 10)) {
+        return Detection::GREY;
+    }
+    return Detection::NO_DETECTION;
+}
+ESPEmotionDetector::Detection ESPEmotionDetector::wait_for_detection(
+    CancellableScope& scope, VideoFeed& feed,
+    std::chrono::seconds timeout
+){
+    Detection last_detection = Detection::NO_DETECTION;
+    size_t confirmations = 0;
+
+    InferenceThrottler throttler(timeout);
+    while (true){
```

Comment:
Any reason you're not using `wait_until()`?

Reply:
I used older (SwSh) programs as a reference since I was originally trying to just look at how color detection worked. Had a look at newer programs and updated it.